src/solver/variable/surveyresults/surveyresults.cpp

```diff
@@ -504,10 +504,16 @@ static inline void WriteIndexHeaderToFileDescriptor(int precisionLevel,
     s += '\n';
 }
 
-uint initializeMaxVariables(uint maxVars, const Data::StudyRuntimeInfos* runtime)
+void SurveyResults::initializeMaxVariables(const Data::Study& s)
 {
+    const auto* runtime = s.runtime;
```

Comment:
Note that getting the max column count here is inefficient : this operation is done for each yearly results and for synthesis results. But it could be done only once.
We should do it elsewhere (see )

Reply:
OK I identified a few steps

1. Compute the matrix size required to store all results
2. Allocate memory
3. Write to that memory
4. Dump that memory into files
5. Deallocate memory

What interests us here is step 1. Currently, it is done both statically and dynamically.

Static: Number of variables with fixed number of columns (mostly 1, sometimes more)
Dynamic: Number of clusters * number of variables by cluster (hard-coded)

Then we take the max of all this to make sure that every result fits (details-xxx.txt, values-xxx.txt, etc.).

For every file, there is a category id (de, de_res, etc.) that's stored in attribute `categoryFileLevel`. Example
```cpp
// src/solver/variable/economy/profitByPlant.h
categoryFileLevel = ResultsType::categoryFile & (Category::de),
// src/solver/variable/economy/dispatchable-generation-margin.h
categoryFileLevel = ResultsType::categoryFile & (Category::id | Category::va),
```

For every category, we need to compute the width & height of the associated matrix for yby & synthesis files.

Why not attach a function to every variable ? It could return 1 (most cases), another constant (thermal/renewable) clusters), or a number depending on the study ("this variable produces one variable per thermal cluster")
```cpp
categoryFileLevel = ResultsType::categoryFile & (Category::de),
columnCalculator = [](const Study& study) { // return the number of thermal clusters };
```

Or, if you don't like lambda functions, we could have a small library of functions (return a constant, return the number of thermal clusters, return the number of inequality binding constraints).

Reply:
- Currently, the computation of max number of columns is done before the clusters or binding constraints are loaded.
So in the state of the current code, it would not work. 
That said, we could change the order in the data load.
- Furthermore, we will have to traverse all output variables, whether we call a classic method or a lambda function, how do you think we could do ?

Reply:
> Currently, the computation of max number of columns is done before the clusters or binding constraints are loaded.

No, see function `initializeMaxVariables` & constructor of class `SurveyResults`

Reply:
> No, see function `initializeMaxVariables` & constructor of class `SurveyResults`
`initializeMaxVariables` is based on information collected before the clusters and BCs are loaded.
`initializeMaxVariables` only gets their value.